src/libs/antares/study/scenario-builder/rules.cpp

```diff
@@ -257,6 +271,41 @@ void Rules::readHydroLevels(const AreaName::Vector& splitKey, String value, bool
     hydroLevels.set(area->index, year, val);
 }
 
+Data::AreaLink* Rules::getLink(const AreaName& areaName, const AreaName& targetAreaName, bool updaterMode)
+{
+    Data::AreaLink* link = study_.areas.findLink(areaName, targetAreaName);
+    if (!link)
+    {
+        // silently ignore the error
+        if (not updaterMode)
+            logs.warning() << "[scenario-builder] The link '" << areaName << " / " << targetAreaName << "' has not been found";
+
+    }
+    return link;
+}
+
+void Rules::readLink(const AreaName::Vector& splitKey, String value, bool updaterMode)
+{
+    const AreaName& areaName = splitKey[1];
+    const AreaName& targetAreaName = splitKey[2];
+    const uint year = splitKey[2].to<uint>();
+
+    Data::Area* area = getArea(areaName, updaterMode);
+    if (!area)
+        return;
+
+    const Data::Area* targetArea = getArea(targetAreaName, updaterMode);
+    if (!targetArea)
+        return;
+
+    AreaLink* link = getLink(areaName, targetAreaName, updaterMode);
+    if (!link)
+        return;
```

Comment:
Variables `targetArea` and `area` are unused.

Reply:
Are they ? There is an **if** statement involving these areas just after their definition.
That said, I can rename them into **fromArea** and **ToArea** --> Done